validator_client/src/cli.rs

```diff
@@ -1,509 +1,498 @@
                     Use * to allow any origin (not recommended in production). \
                     If no value is supplied, the CORS allowed origin is set to the listen \
-                    address of this server (e.g., http://localhost:5064).")
-                .action(ArgAction::Set)
-                .display_order(0)
-        )
-        .arg(
-            Arg::new("enable-high-validator-count-metrics")
-                .long("enable-high-validator-count-metrics")
-                .help("Enable per validator metrics for > 64 validators. \
-                    Note: This flag is automatically enabled for <= 64 validators. \
-                    Enabling this metric for higher validator counts will lead to higher volume \
-                    of prometheus metrics being collected.")
-                .action(ArgAction::SetTrue)
-                .help_heading(FLAG_HEADER)
-                .display_order(0)
-        )
-        /*
-         * Explorer metrics
-         */
-         .arg(
-            Arg::new("monitoring-endpoint")
-                .long("monitoring-endpoint")
-                .value_name("ADDRESS")
-                .help("Enables the monitoring service for sending system metrics to a remote endpoint. \
+                    address of this server (e.g., http://localhost:5062).",
+        display_order = 0
+    )]
+    pub http_allow_origin: Option<String>,
+
+    #[clap(
+        long,
+        requires = "http",
+        help = "If present, allow access to the DELETE /lighthouse/keystores HTTP \
+                API method, which allows exporting keystores and passwords to HTTP API \
+                consumers who have access to the API token. This method is useful for \
+                exporting validators, however it should be used with caution since it \
+                exposes private key data to authorized users.",
+        display_order = 0,
+        help_heading = FLAG_HEADER
+    )]
+    pub http_allow_keystore_export: bool,
+
+    #[clap(
+        long,
+        requires = "http",
+        help = "If present, any validators created via the HTTP will have keystore \
+                passwords stored in the secrets-dir rather than the validator \
+                definitions file.",
+        display_order = 0,
+        help_heading = FLAG_HEADER
+    )]
+    pub http_store_passwords_in_secrets_dir: bool,
+
+    /* Prometheus metrics HTTP server related arguments */
+    #[clap(
+        long,
+        help = "Enable the Prometheus metrics HTTP server. Disabled by default.",
+        display_order = 0,
+        help_heading = FLAG_HEADER
+    )]
+    pub metrics: bool,
+
+    #[clap(
+        long,
+        value_name = "ADDRESS",
+        default_value_if("metrics", ArgPredicate::IsPresent, "127.0.0.1"),
+        help = "Set the listen address for the Prometheus metrics HTTP server.",
+        display_order = 0
+
+    )]
+    pub metrics_address: IpAddr,
```

Comment:
For some reason, the CLI thinks that `--metrics-address` is a required flag, even when `--metrics` is provided.